solr/core/src/java/org/apache/solr/cloud/ZkCLI.java

```diff
@@ -327,6 +355,22 @@ public static void main(String[] args) throws InterruptedException,
           }
           byte [] data = zkClient.getData(arglist.get(0).toString(), null, null, true);
           stdout.println(new String(data, StandardCharsets.UTF_8));
+        } else if (line.getOptionValue(CMD).equalsIgnoreCase(GETCOMPRESSED)) {
+          @SuppressWarnings({"rawtypes"})
+          List arglist = line.getArgList();
+          if (arglist.size() != 1) {
+            stdout.println("-" + GET + " requires one arg - the path to get");
+            System.exit(1);
+          }
+          byte [] data = zkClient.getData(arglist.get(0).toString(), null, null, true);
+          try {
+            CompressorInputStream input = new CompressorStreamFactory()
+                .createCompressorInputStream(new ByteArrayInputStream(data));
+            byte[] decompressedBytes = org.apache.commons.compress.utils.IOUtils.toByteArray(input);
+            stdout.println(new String(decompressedBytes, StandardCharsets.UTF_8));
+          } catch (IOException | CompressorException compressorException) {
+            throw new RuntimeException(compressorException);
+          }
```

Comment:
Is there a way to avoid having a distinct "getcompressed" command? iirc Ishan mentioned something about this: determining based on the first few bytes whether the content is compressed or not, and decompressing on the fly if needed?

This would be particularly practical when rolling this out on a live cluster.

Reply:
mmm maybe nevermind? What I was suggesting appears to be implemented in `ClusterState.java` (I'm mystified how I overlooked that initially), and maybe in zkcli there's a reason to support the ability to support making the request more explicit? (though at first glance I'd think maybe zkcli would want also to be dynamic?

Reply:
I just made an update to improve this. I pushed the compression check and decompression down into SolrZkClient.getData() so that all consumers, including ZkCLI and Zookeper admin calls can use that dynamically.

My sense is that the check for compression is lightweight enough to not add significant extra work at that level even applied to all getData() calls.